agent/api/errors.go

```diff
@@ -25,6 +25,11 @@ func IsInstanceTypeChangedError(err awserr.Error) bool {
 	return strings.Contains(err.Message(), INSTANCE_TYPE_CHANGED_ERROR_MESSAGE)
 }
 
+func IsInvalidAttributeError(err awserr.Error) bool {
```

Comment:
We need a real error type if we're going to do this.  String matching on the human-readable error message is too fragile.